### ISSUE TITLE
`properties(relationship)`throws Neo.DatabaseError.Statement.ExecutionFailed in a composite database

### ISSUE THREAD

**Issue by rlch** [neo4j/neo4j#13028]
- Neo4j version: 5.2.0
- Operating system: Ubuntu
- API/Driver: Cypher
 
- **Steps to reproduce**
1. Create a composite database
2. Create relationship `r` with no properties
3. Return `properties(r)` using composite database

## Logs

```cypher
USE bridge.prod
MATCH (n:QuestionGroup)-[e:CHALLENGES]->(v:Skill) RETURN properties(e) LIMIT 25
```

```bash
2023-01-13 12:49:33.835+0000 INFO  [o.n.b.p.c.c.n.SocketNettyConnector] Bolt enabled on 0.0.0.0:7687.
2023-01-13 12:49:33.843+0000 INFO  [o.n.b.p.c.c.n.SocketNettyConnector] Bolt (Routing) enabled on 0.0.0.0:7688.
2023-01-13 12:49:33.843+0000 INFO  [o.n.b.BoltServer] Bolt server started
2023-01-13 12:49:33.845+0000 INFO  [o.n.s.A.ServerComponentsLifecycleAdapter] Starting web server
2023-01-13 12:49:34.219+0000 INFO  [c.n.c.d.d.TopologyState] 'ServerId{ef02613d}' information change: 'ServerInfo[versionInfo=5.2.0, connectors={BOLT=localhost:7687, RAFT=53cf0cee2be3:7000, CLUSTER=53cf0cee2be3:6000}, serverConfig={initial.server.allowed_databases, a ',' separated set with elements of type 'a string'.=[], initial.server.mode_constraint, one of [PRIMARY, SECONDARY, NONE]=NONE, server.groups, a ',' separated list with elements of type 'a string identifying a Server Tag'.=[], server.cluster.initial_mode_constraint, a string=NONE, initial.server.denied_databases, a ',' separated set with elements of type 'a string'.=[]}]' -> 'ServerInfo[versionInfo=5.2.0, connectors={RAFT=53cf0cee2be3:7000, BOLT=localhost:7687, CLUSTER=53cf0cee2be3:6000, INTRA_BOLT=localhost:7688}, serverConfig={initial.server.allowed_databases, a ',' separated set with elements of type 'a string'.=[], initial.server.mode_constraint, one of [PRIMARY, SECONDARY, NONE]=NONE, server.groups, a ',' separated list with elements of type 'a string identifying a Server Tag'.=[], server.cluster.initial_mode_constraint, a string=NONE, initial.server.denied_databases, a ',' separated set with elements of type 'a string'.=[]}]'
2023-01-13 12:49:36.133+0000 INFO  [c.n.s.e.EnterpriseNeoWebServer] Remote interface available at http://localhost:7474/
2023-01-13 12:49:36.134+0000 INFO  [o.n.s.A.ServerComponentsLifecycleAdapter] Web server started.
2023-01-13 12:49:36.150+0000 INFO  [o.n.g.f.DatabaseManagementServiceFactory] id: 1FBF6A9B00BD140DB10970FBBE78B5F92899DC08D498B2CAF7858AC255540382
2023-01-13 12:49:36.151+0000 INFO  [o.n.g.f.DatabaseManagementServiceFactory] name: system
2023-01-13 12:49:36.175+0000 INFO  [o.n.g.f.DatabaseManagementServiceFactory] creationDate: 2022-11-28T12:03:46.987Z
2023-01-13 12:49:36.289+0000 INFO  [c.n.c.d.d.TopologyState] 'ServerId{ef02613d}' information change: 'ServerInfo[versionInfo=5.2.0, connectors={RAFT=53cf0cee2be3:7000, BOLT=localhost:7687, CLUSTER=53cf0cee2be3:6000, INTRA_BOLT=localhost:7688}, serverConfig={initial.server.allowed_databases, a ',' separated set with elements of type 'a string'.=[], initial.server.mode_constraint, one of [PRIMARY, SECONDARY, NONE]=NONE, server.groups, a ',' separated list with elements of type 'a string identifying a Server Tag'.=[], server.cluster.initial_mode_constraint, a string=NONE, initial.server.denied_databases, a ',' separated set with elements of type 'a string'.=[]}]' -> 'ServerInfo[versionInfo=5.2.0, connectors={RAFT=53cf0cee2be3:7000, HTTP=localhost:7474, BOLT=localhost:7687, CLUSTER=53cf0cee2be3:6000, INTRA_BOLT=localhost:7688}, serverConfig={initial.server.allowed_databases, a ',' separated set with elements of type 'a string'.=[], initial.server.mode_constraint, one of [PRIMARY, SECONDARY, NONE]=NONE, server.groups, a ',' separated list with elements of type 'a string identifying a Server Tag'.=[], server.cluster.initial_mode_constraint, a string=NONE, initial.server.denied_databases, a ',' separated set with elements of type 'a string'.=[]}]'
2023-01-13 12:49:38.264+0000 WARN  [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=214, gcTime=0, gcCount=0}
2023-01-13 12:52:39.852+0000 ERROR [o.n.b.p.c.f.StateMachineSPIImpl] Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: Expected positive int value, got 0, reference 4870f7b4-a7df-4350-a8c1-b198ca5227d5.
2023-01-13 12:52:39.857+0000 ERROR [o.n.b.p.c.f.StateMachineSPIImpl] Client triggered an unexpected error [Neo.DatabaseError.Statement.ExecutionFailed]: Expected positive int value, got 0, reference 4870f7b4-a7df-4350-a8c1-b198ca5227d5.
org.neo4j.fabric.executor.FabricException: Expected positive int value, got 0
	at org.neo4j.fabric.executor.Exceptions.transform(Exceptions.java:56) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.executor.Exceptions.transform(Exceptions.java:32) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:59) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.bolt.BoltQueryExecutionImpl$QueryExecutionImpl.request(BoltQueryExecutionImpl.java:151) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.transaction.AbstractCypherAdapterStream.handleRecords(AbstractCypherAdapterStream.java:96) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.transaction.result.ResultHandler.onPullRecords(ResultHandler.java:64) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.v40.messaging.result.PullResultConsumer.consume(PullResultConsumer.java:39) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.transaction.TransactionStateMachine$State.consumeResult(TransactionStateMachine.java:517) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.transaction.TransactionStateMachine$State$2.streamResult(TransactionStateMachine.java:370) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.transaction.TransactionStateMachine.streamResult(TransactionStateMachine.java:107) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.transaction.StatementProcessorTxManager.streamResults(StatementProcessorTxManager.java:226) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.transaction.StatementProcessorTxManager.pullData(StatementProcessorTxManager.java:117) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.v40.fsm.InTransactionState.processStreamPullResultMessage(InTransactionState.java:73) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.v40.fsm.AbstractStreamingState.processUnsafe(AbstractStreamingState.java:47) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.v40.fsm.InTransactionState.processUnsafe(InTransactionState.java:60) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.v40.fsm.FailSafeState.process(FailSafeState.java:44) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.fsm.AbstractStateMachine.nextState(AbstractStateMachine.java:136) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.fsm.AbstractStateMachine.process(AbstractStateMachine.java:96) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.connector.connection.AtomicSchedulingConnection.lambda$submit$4(AtomicSchedulingConnection.java:113) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.connector.connection.AtomicSchedulingConnection.executeJob(AtomicSchedulingConnection.java:336) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.connector.connection.AtomicSchedulingConnection.doExecuteJobs(AtomicSchedulingConnection.java:270) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at org.neo4j.bolt.protocol.common.connector.connection.AtomicSchedulingConnection.executeJobs(AtomicSchedulingConnection.java:212) ~[neo4j-bolt-5.2.0.jar:5.2.0]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IllegalArgumentException: Expected positive int value, got 0
	at org.neo4j.util.Preconditions.requirePositive(Preconditions.java:86) ~[neo4j-common-5.2.0.jar:5.2.0]
	at org.neo4j.values.virtual.MapValueBuilder.capacity(MapValueBuilder.java:62) ~[neo4j-values-5.2.0.jar:5.2.0]
	at org.neo4j.values.virtual.MapValueBuilder.<init>(MapValueBuilder.java:38) ~[neo4j-values-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.QuerySubject$TaggingQuerySubject.withTaggedId(QuerySubject.java:296) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.QuerySubject$TaggingQuerySubject.withTaggedId(QuerySubject.java:262) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.QuerySubject$TaggingQuerySubject.onField(QuerySubject.java:240) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.codegen.OperatorTaskPipeline0_21.compiledOperate(Unknown Source) ~[?:?]
	at org.neo4j.cypher.internal.runtime.pipelined.operators.CompiledTask.operateWithProfile(OperatorCodeGenBaseTemplates.scala:680) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeOperators(PipelineTask.scala:81) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.PipelineTask.executeWorkUnit(PipelineTask.scala:65) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker$.executeTask(Worker.scala:971) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker.executeTask(Worker.scala:120) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker$.workOnQuery(Worker.scala:874) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.Worker$.workOnQuery(Worker.scala:317) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.runtime.pipelined.execution.CallingThreadExecutingQuery.request(CallingThreadExecutingQuery.scala:43) ~[neo4j-cypher-pipelined-runtime-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.PipelinedRuntimeResult.request(PipelinedRuntimeResult.scala:79) ~[neo4j-enterprise-cypher-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:90) ~[neo4j-cypher-5.2.0.jar:5.2.0]
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:149) ~[neo4j-cypher-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.doRequest(QuerySubject.java:155) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.QuerySubject$BasicQuerySubject$1.request(QuerySubject.java:143) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.request(Operators.java:2158) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.FluxPeek$PeekSubscriber.request(FluxPeek.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at reactor.core.publisher.StrictSubscriber.request(StrictSubscriber.java:138) ~[reactor-core-3.4.24.jar:3.4.24]
	at org.neo4j.fabric.stream.Rx2SyncStream$RecordSubscriber.request(Rx2SyncStream.java:109) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.Rx2SyncStream.maybeRequest(Rx2SyncStream.java:77) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	at org.neo4j.fabric.stream.Rx2SyncStream.readRecord(Rx2SyncStream.java:45) ~[neo4j-fabric-5.2.0.jar:5.2.0]
	... 24 more

```

**Comment by mnd999** [comment]
Thanks for the report, that looks like a bug. We'll look into it and get back to you.

**Comment by LinneaAndersson** [comment]
@rlch Thank you for reporting this, the bug has now been fixed and will be released in neo4j version 5.5. 
Let us know if you have any further questions.
